numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -521,7 +527,13 @@ wrap_aligned_contig_transfer_function(
                 NPY_LOWLEVEL_BUFFER_BLOCKSIZE*src_itemsize;
 
     /* Set the function and data */
-    if (init_dest) {
+    if (init_dest && out_needs_api) {
```

Comment:
nit: This block might be clearer as:
```C
    if (init_dest) {
        if(out_needs_api) {
            *out_stransfer = &_strided_to_strided_contig_align_wrap_init_dest_safe;
        }
        else {
            *out_stransfer = &_strided_to_strided_contig_align_wrap_init_dest;
        }
    }
    else {
        if (out_needs_api) {
            *out_stransfer = &_strided_to_strided_contig_align_wrap_safe;
        }
        else {
           *out_stransfer = &_strided_to_strided_contig_align_wrap;
        }
    }
```
or perhaps
```C
*out_stransfer = init_dest ? out_needs_api ? &_strided_to_strided_contig_align_wrap_init_dest_safe
                                           : &_strided_to_strided_contig_align_wrap_init_dest
                           : out_needs_api ? &_strided_to_strided_contig_align_wrap_safe
                                           : &_strided_to_strided_contig_align_wrap;
```
Let's see what other reveiwers think

Reply:
Or perhaps the real answer here is to add two fields to `_align_wrap_data`, and then all the branches dissappear.

Reply:
Part of the reason I wrote it this way was because the deep nesting and giant function names result in long lines (which are hard to wrap since they're each just an assignment). One option could just be to rename the functions...

I like the idea of adding them to `_align_wrap_data` too, but I'll defer to your judgement here.

Reply:
To me adding to `_align_wrap_data` makes the most sense

Reply:
That's nice, because it then makes a lot of other code disappear. I'll have something up soon.